docs/user-guide/coordinate-transformations.ipynb

```diff
@@ -0,0 +1,424 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "239c7ad4-44b0-4d8f-8dcb-6fa5833abc45",
+   "metadata": {},
+   "source": [
+    "# Coordinate transformations\n",
+    "\n",
+    "## Motivation\n",
+    "\n",
+    "In all fields of science we frequently encounter data that is represented in coordinates or coordinate systems that are not apt for certain operations or visualizations.\n",
+    "In these cases we may thus need to compute new coordinates based on one or multiple existing coordinates.\n",
+    "For simple cases this may just be done by hand.\n",
+    "Consider:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8b11dfb5-f9b9-43aa-8e56-418bcc0489a7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import scipp as sc\n",
+    "x = sc.linspace(dim='x', unit='m', start=1., stop=55., num=100)\n",
+    "da = sc.DataArray(data=x*x, coords={'x':x})\n",
+    "da.plot(figsize=(4,3))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "43d74a82-1f9c-48e3-a587-2d932288b532",
+   "metadata": {},
+   "source": [
+    "We may want to use $x^2$ instead of $x$ as a coordinate, do highlight the quadratic nature of our data:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4ad57397-8d6c-46ac-a748-99207697734b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "da2 = da.copy()\n",
+    "da2.coords['x_square'] = x * x\n",
+    "da2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "15677625-8898-40c7-a3fd-2d0c4395d612",
+   "metadata": {},
+   "source": [
+    "While adding a new coordinate may often be done with a single line of code, the above example highlights the first shortcoming of this approach:\n",
+    "To actually visualize `da` using this new coordinate we must additionally rename the dimension:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7b222310-40d6-4b95-92dd-9c2e7b7bcb63",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "da2 = da2.rename_dims({'x':'x_square'})\n",
```

Comment:
```suggestion
    "da2 = da2.rename_dims({'x': 'x_square'})\n",
```
Can yapf and flake8 process notebooks?

Reply:
This seems useful, will try it out: https://jupyterlab-code-formatter.readthedocs.io/en/latest/installation.html#installation-step-1-installing-the-plugin-itself